annotations/builder/src/main/java/io/sundr/builder/internal/functions/ToMethod.java

```diff
@@ -813,6 +817,14 @@ private Statement createAddToDescendantsFallback(String type, String name) {
               "else {  VisitableBuilder<? extends " + type + ",?> builder = builderOf(item); _visitables.get(\"" + name
                   + "\").add(builder);this." + name + ".add(builder); }");
         }
+
+        private StringStatement createAddOrSetIndex(String op, String propertyName, String returnType) {
+          return new StringStatement(
+              "if (index < 0 || index >= " + propertyName + ".size()) { _visitables.get(\"" + propertyName
```

Comment:
Can you please elaborate here?
Why do we need fallbacks? Shouldn't we just let it fail instead?

Reply:
> Why do we need fallbacks? Shouldn't we just let it fail instead?

The fallbacks, via builderOf, handles the failure case.  Several these methods had the same problem as #350

Reply:
I am still not sure I follow. Can you please further elaborate?
Given that we also have an additional topic regarding Editables in #376 it might make sense to schedule a meeting or something.

Reply:
> I am still not sure I follow. Can you please further elaborate?

The existing implementation of setToCollection(index, interface) looks like:

```
if (item instanceof Concrete1) { setToConcrete...
if (item instanceof Concrete2) { setToConcrete...

return (A)this;
```

So if the item is a not a known buildable type it will simply fall through and do nothing - which is what was observed with #350

This change makes setTo and addTo have the else case

} else { VisitableBuilder builder = builderOf(item) ... }

So that if something has it builder it will succeed, or if not there will be an exception.